### PR TITLE
Add no-theme-usage rule

### PR DESCRIPTION
**Pull request by tjunyamasaki**
This PR adds `no-theme-usage` rule. It blocks MUI's `theme` usage, asking the user to replace it with cozy's tokens.

### CODE REVIEW

**Review by danielasy — APPROVED** [review]
Good test suite!

File: eslint-plugin-quintoandar/rules/no-theme-usage.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const reportText = `
```
Comment (fernahh) [inline eslint-plugin-quintoandar/rules/no-theme-usage.js:1]: What do you think about adding a reference for cozy-theme documentation?

(cc @thayllachristine)

**Review by fernahh — APPROVED** [review]

File: eslint-plugin-quintoandar/rules/no-theme-usage.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          node.object &&
+          isId(node.object.type) &&
+          node.object.name === "theme"
+        ) {
+          context.report({
+            node,
+            message: reportText,
+          });
+        }
+      }
```
[the remark this answers is not in the document]
Comment (tjunyamasaki) [reply eslint-plugin-quintoandar/rules/no-theme-usage.js:31]: Nice! I will change it :smile:

File: eslint-plugin-quintoandar/rules/no-theme-usage.js (hunk 1 above)
Comment (tjunyamasaki) [reply eslint-plugin-quintoandar/rules/no-theme-usage.js:1]: It could be good but this repository is open source, is this a problem?

Comment (fernahh) [reply eslint-plugin-quintoandar/rules/no-theme-usage.js:1]: Yes, it's a problem :grimacing:

**Pull request merged**